### PR TITLE
Add Get-GkeCluster cmdlet

### PR DESCRIPTION
**Pull request by quoctruong**
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-powershell/494)

### CODE REVIEW

File: Google.PowerShell/Container/GkeClusterCmdlets.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+namespace Google.PowerShell.Container
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Gets Google Container Clusters.
+    /// </para>
+    /// <para type="description">
+    /// Gets Google Container Clusters. If -Project parameter is not specified, the default project will be used.
+    /// If -Zone or -ClusterName is not used, the cmdlet will return every clusters in every zone in the project.
```
Comment (ILMTitan) [inline Google.PowerShell/Container/GkeClusterCmdlets.cs]: "If neither -Zone nor -ClusterName is used" is better. It is too easy to skip over the "not".

File: Google.PowerShell/Container/GkeClusterCmdlets.cs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    ///   <para>Gets the cluster "my-cluster" in the default zone of the default project.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GkeCluster -ClusterName "my-cluster", "my-cluster-2" -Zone "us-central1-a"</code>
+    ///   <para>
+    ///   Gets the cluster "my-cluster", "my-cluster-2" in the zone "us-central1-a" of the default project.
+    ///   </para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/compute/docs/reference/latest/machineTypes#resource)">
+    /// [Machine Type resource definition]
```
Comment (ILMTitan) [inline Google.PowerShell/Container/GkeClusterCmdlets.cs]: This link looks like a copy-paste error.
Should it be https://cloud.google.com/container-engine/reference/rest/v1/projects.zones.clusters#Cluster?

File: Google.PowerShell/Container/GkeClusterCmdlets.cs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            ProjectsResource.ZonesResource.ClustersResource.ListRequest listRequest =
+                Service.Projects.Zones.Clusters.List(project, zone);
+            ListClustersResponse response = listRequest.Execute();
+            if (response.Clusters != null)
+            {
+                foreach (Cluster cluster in response.Clusters)
+                {
+                    yield return cluster;
+                }
+            }
```
Comment (ILMTitan) [inline Google.PowerShell/Container/GkeClusterCmdlets.cs]: Perhaps add a warning about missing zones? Do we understand what missing zones means?

**Comment by quoctruong** [comment]
@ILMTitan PTAL

File: Google.PowerShell/Container/GkeClusterCmdlets.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+namespace Google.PowerShell.Container
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Gets Google Container Clusters.
+    /// </para>
+    /// <para type="description">
+    /// Gets Google Container Clusters. If -Project parameter is not specified, the default project will be used.
+    /// If neither -Zone nor -ClusterName is used, the cmdlet will return every clusters in every zone in the project.
```
Comment (ahandley) [inline Google.PowerShell/Container/GkeClusterCmdlets.cs]: "return every clusters" -> "return every cluster"

Repeated once below as well.

File: Google.PowerShell/Container/GkeClusterCmdlets.cs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    ///   <para>Lists all container clusters in zone us-central1-a for the project "my-project".</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GkeCluster -ClusterName "my-cluster"</code>
+    ///   <para>Gets the cluster "my-cluster" in the default zone of the default project.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GkeCluster -ClusterName "my-cluster", "my-cluster-2" -Zone "us-central1-a"</code>
+    ///   <para>
+    ///   Gets the cluster "my-cluster", "my-cluster-2" in the zone "us-central1-a" of the default project.
```
Comment (ahandley) [inline Google.PowerShell/Container/GkeClusterCmdlets.cs]: "Gets the cluster" -> "Gets the clusters"

Comment (quoctruong) [inline Google.PowerShell/Container/GkeClusterCmdlets.cs]: Fixed.

File: Google.PowerShell/Container/GkeClusterCmdlets.cs (hunk 4 above)
Comment (quoctruong) [inline Google.PowerShell/Container/GkeClusterCmdlets.cs]: Fixed.

File: Google.PowerShell/Container/GkeClusterCmdlets.cs [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+                {
+                    yield return cluster;
+                }
+            }
+            // The list of clusters returned may be missing these zones.
+            if (response.MissingZones != null)
+            {
+                foreach (string missingZone in response.MissingZones)
+                {
+                    WriteWarning(missingZone);
```
Comment (ILMTitan) [inline Google.PowerShell/Container/GkeClusterCmdlets.cs]: We need to add some human readable context to this list of missing clusters. Writing a warning of "us-central1-f" is not very informative.

**Comment by quoctruong** [comment]
@ILMTitan PTAL

**Comment by quoctruong** [comment]
Thanks!

**Pull request merged**